### ISSUE TITLE
What does "content" mean in SC 1.4.10 Reflow

### ISSUE THREAD

**Issue by JAWS-test** [w3c/wcag#2073]
1.4.10 Reflow specifies that "content" may not be scrolled two-dimensionally at 320px width or 265px height (with a few exceptions).

But what is "content" in the context of 1.4.10?
1. the entire page
2. a contiguous block of text
3. a paragraph of text
4. a visual page area that could be marked up with a landmark region, for example (e.g., the main menu in the navigation region)
5. a column in the page layout
6. something completely different?

The answer seems important to me, as it seems that a page is allowed to have several different contents that have different scrolling directions (see https://github.com/w3c/wcag/issues/2063#issuecomment-932797846 and https://github.com/w3c/wcag/issues/2063#issuecomment-932644312)

A few examples where I don't know if 1.4.10 is met or not:
1. the menu must be scrolled only horizontally, the rest of the page only vertically.
2. a text with 2 columns, in each column I have to scroll only vertically, but to get to the next column I have to scroll horizontally
3. text that needs to be scrolled horizontally at 320px, but not vertically, although the text is horizontal (e.g. a text in English language)
4. the same text as in example 3, but it has to be scrolled horizontally already at 1280px, but not vertically (and at 320px as well, although with more scrolling effort due to the 400% zoom)
5. a form with 2 columns, which has to be scrolled vertically and horizontally, but horizontally only to get to the 2nd column (the filling direction in the form is from top to bottom, then to the 2nd column).
6. a form with 2 columns, which has to be scrolled vertically and horizontally, but horizontally only to get to the 2nd column (but the filling direction in the form is from left to right, i.e. you have to switch constantly between left and right)
7. the same form as in 6. but it must be scrolled already at 1280px between the two columns

There is a [definition of "content" in WCAG glossary](https://www.w3.org/TR/WCAG21/#dfn-content) which in my opinion, however, does not help in answering the question:

> information and sensory experience to be communicated to the user by means of a user agent, including code or markup that defines the content's structure, presentation, and interactions

**Comment by alastc** [comment]
> But what is "content" in the context of 1.4.10?

In the context of WCAG, everything within the page.

> 1. the menu must be scrolled only horizontally, the rest of the page only vertically.

If a chunk of content (e.g. a menu) scrolls horizontally, it should meet the second bullet on vertical height.

> 2. a text with 2 columns, in each column I have to scroll only vertically, but to get to the next column I have to scroll horizontally

If the entire page is scrolling in both directions, that's going to fail. It's an odd and problematic case (as discussed elsewhere), but caught by the wording.

> 3. text that needs to be scrolled horizontally at 320px, but not vertically, although the text is horizontal (e.g. a text in English language)

If it's scrolling horizontally, it would need to meet the 256px height bullet.

> 4. the same text as in example 3, but it has to be scrolled horizontally already at 1280px, but not vertically (and at 320px as well, although with more scrolling effort due to the 400% zoom)

Same as 3.

> a form with 2 columns, which has to be scrolled vertically and horizontally, but horizontally only to get to the 2nd column (the filling direction in the form is from top to bottom, then to the 2nd column).

Same as 2.

> 5. a form with 2 columns, which has to be scrolled vertically and horizontally, but horizontally only to get to the 2nd column (but the filling direction in the form is from left to right, i.e. you have to switch constantly between left and right)

Same as 2.

>  6. the same form as in 6. but it must be scrolled already at 1280px between the two columns

Same as 2.

**Comment by alastc** [comment]
> is "content" now the whole page or a part of the page (because you write about "content" and "chunk of content").

I wonder if there is a language issue here? Let me try a metaphor: Consider the page as a bucket, and the content is water in the bucket. 

You can talk about the content as a whole, all the water in the bucket. Or you could talk about bits of it, like the top 1/3rd of the water in the bucket. It is all content (water), but you can take it as a whole or as parts. 

> A page that has a vertical and horizontal scrollbar on the outside is always wrong (as long as none of the exceptions like table etc.). 

Paraphrasing the SC text: "Content can be presented without loss of information or functionality, and without requiring scrolling in two dimensions ... Except for **parts of the content** which require two-dimensional layout for usage or meaning."

So the exception applies to **parts of the content**. Therefore a table (wrapper) can have scrolling, but that exception doesn't apply to the rest of the content on the page.

That's why I said if the whole page has scrolling (e.g. at 320px wide) that would be a fail.

> A page that has only an outer vertical scrollbar is not wrong, even if page areas have inner horizontal scrollbars (as long as those page areas don't have vertical scrollbars).

Agree. A page which scrolls vertically can have inner areas which have horizontally scrolling (and fit 250px height).

**Comment by alastc** [comment]
Jon wrote: 

>  I think we need to have the group vote on their interpretation (if it hasn't already occurred) because there doesn't appear to be consensus on this.

Ok, I'll raise a question. Drafting the question, I think it would be:
<hr>

A question has been raised in #2073 about whether having scrolling in horizontal and vertical directions would always be a fail of 1.4.10 Reflow. The scenario would when: "a page has 2 columns, and for each column a user can scroll vertically, but to get to the next column you have to scroll horizontally."

Does that scenario fail 1.4.10? (Please comment with reasoning)

**Comment by mraccess77** [comment]
If I understand @alastc comment - In the example of video streaming site - you could have horizontally scrolling widgets with videos (as long as they are not > 256px high) and you could also have page level vertical scrolling.  But you could not have page level horizontal scrolling and pass.    Is that what you said?

I'd assume that if there was a horizontal scrollbar that appears for the page but it didn't actually need to be used that it would be ok.  I see this sometimes with pages the scrollbar appears but it is not needed as there might just be some extra padding on the side causing it.

**Comment by JAWS-test** [comment]
Let's take a concrete example to illustrate my question: https://codepen.io/jaws-test/pen/gOxboYW

Would this page be ok because it has 3 different levels of content:
* Level 1: the whole page, is only scrolled vertically
* Level 2: the only horizontally scrollable parts of the page (with grey background), like Netflix (except that Netflix has videos here and no text blocks).
* Level 3: the text blocks (with black border) that are only scrolled horizontally.

Result: 
* I can read every text block without horizontal scrolling.
* To move from text block to text block, I have to scroll vertically and horizontally.
* But I never have to scroll down, then to the right and then up again.

**Comment by bruce-usab** [comment]
@JAWS-test the concrete examples are appreciated.  The WCAG2 Issues TF will continue to iterate on guidance on reflow.  We discussed _very briefly_ a couple of your examples.

> Let's take a concrete example to illustrate my question: https://codepen.io/jaws-test/pen/gOxboYW

That's a pass.  It is essentially the same scenario as a two-dimensional plane of thumbnails, but only using text.  See the example video selection page, in [Alastair's comment on issue 3378](https://github.com/w3c/wcag/issues/3378#issuecomment-1766220849).

> Now, on the other hand, the example https://codepen.io/jaws-test/pen/xxLbpZb, in which only the text blocks are higher.

That's a fail.  A set of narrow columns means lots of vertical scrolling while reading one column, then horizontal scrolling when trying to find the next column.

> By the way, all three examples do not violate the initial intention of SC 1.4.10

I do not agree.  The intention is to support fluid reading even with significant magnification.  Your bulleted results reflect that.

**Comment by mraccess77** [comment]
Regarding columns/cells of text - maybe the difference here is that the columns are taller than 256 CSS pixels? We need to find a way to define when columns pass and when they fail.

**Comment by mitchellevan** [comment]
> We need to find a way to define when columns pass and when they fail.

In a [review of PR 3695](https://github.com/w3c/wcag/pull/3695#pullrequestreview-1943603398) I've proposed applying SC 1.4.10 to each "[section](https://www.w3.org/TR/WCAG/#dfn-section)" as defined in WCAG. 

Consider this scenario. A page has three tall columns of left-to-right text, such as English. The columns appear side-by-side regardless of the viewport size. Each column is 320 CSS pixels wide. (In other words, the page is not mobile responsive, but each column is narrow.) In this scenario:

* If each [section](https://www.w3.org/TR/WCAG/#dfn-section) of content fits within a single column, then it would pass 1.4.10 under my proposed interpretation.
* But if a [section](https://www.w3.org/TR/WCAG/#dfn-section) runs to two or more columns, then it would fail 1.4.10 under my proposed interpretation.

**Comment by mitchellevan** [comment]
@mbgower's [comment in 3599](https://github.com/w3c/wcag/issues/3599#issuecomment-2009881043) is relevant. Michael's comment appears to support PR #3695.

@scottaohara's [comment in 3599 with link to a demo](https://github.com/w3c/wcag/issues/3599#issuecomment-2096638337) is relevant. Scott's demo appears similar to the scenarios described in the current issue.

**Comment by scottaohara** [comment]
closing as resolved by the merging of 4055's update to the reflow understanding doc.  if any questions remain, please file a new issue.
